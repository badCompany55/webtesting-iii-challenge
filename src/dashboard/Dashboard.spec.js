// Test away
import React from "react";
import { render, fireEvent, cleanup } from "react-testing-library";
import Dashboard from "./Dashboard.js";

afterEach(cleanup);

describe("<Dashboard/>", () => {
  it("should render without crashing", () => {
    render(<Dashboard />);
  });

  it("should change button text on click", () => {
    const { getByText, getByTestId } = render(<Dashboard />);
    const button1 = getByTestId("unlockLock");
    const button2 = getByTestId("openClose");

    fireEvent.click(button2);
    expect(button2.textContent).toBe("Open Gate");
    fireEvent.click(button1);
    expect(button1.textContent).toBe("Unlock Gate");
    fireEvent.click(button2);
    expect(button2.textContent).toBe("Open Gate");
    expect(button2.textContent).not.toBe("Close Gate");
    fireEvent.click(button1);
    expect(button1.textContent).toBe("Lock Gate");
    fireEvent.click(button2);
    expect(button2.textContent).toBe("Close Gate");
    fireEvent.click(button1);
    expect(button1.textContent).toBe("Lock Gate");
  });
});
