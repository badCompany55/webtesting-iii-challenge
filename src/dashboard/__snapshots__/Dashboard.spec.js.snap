// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> Should match snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="unlockLock"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="openClose"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
